permissions/resource_access_control_rule_set.go

```diff
@@ -0,0 +1,139 @@
+	}
+	fetchLatestEtagAndUpdateRuleSet := func(ctx context.Context, c *common.DatabricksClient, ruleSetUpdateReq iam.UpdateRuleSetRequest) (string, error) {
+		ruleSetGetRes, err := readFromWsOrAcc(ctx, c, iam.GetRuleSetRequest{
+			Name: ruleSetUpdateReq.Name,
+			Etag: "",
+		})
+		if err != nil {
+			return "", err
+		}
+		ruleSetUpdateReq.RuleSet.Etag = ruleSetGetRes.Etag
+		ruleSetUpdateRes, err := updateThroughWsOrAcc(ctx, c, ruleSetUpdateReq)
+		if err != nil {
+			return "", err
+		}
+		return ruleSetUpdateRes.Etag, nil
+	}
+	handleConflictAndUpdate := func(ctx context.Context, c *common.DatabricksClient, ruleSetUpdateReq iam.UpdateRuleSetRequest) (string, error) {
+		ruleSetUpdateRes, err := updateThroughWsOrAcc(ctx, c, ruleSetUpdateReq)
+		if err != nil {
+			var aerr *apierr.APIError
+			if !errors.As(err, &aerr) {
+				return "", err
+			}
+			if aerr.StatusCode == http.StatusConflict {
+				if aerr.ErrorCode == "RESOURCE_CONFLICT" {
+					// we need to get and update
+					etag, err := fetchLatestEtagAndUpdateRuleSet(ctx, c, ruleSetUpdateReq)
+					return etag, err
+				}
+			}
+			return "", err
+		}
+		return ruleSetUpdateRes.Etag, err
+	}
+	return common.Resource{
+		Schema: s,
+		Create: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			var ruleSetUpdateReq iam.UpdateRuleSetRequest
+			common.DataToStructPointer(d, s, &ruleSetUpdateReq.RuleSet)
+			ruleSetUpdateReq.Name = ruleSetUpdateReq.RuleSet.Name
+			etag, err := fetchLatestEtagAndUpdateRuleSet(ctx, c, ruleSetUpdateReq)
+			if err != nil {
+				return err
+			}
+			d.Set("etag", etag)
+			d.SetId(ruleSetUpdateReq.Name)
+			return nil
+		},
+		Read: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			data, err := readFromWsOrAcc(ctx, c, iam.GetRuleSetRequest{
+				Name: d.Id(),
+				Etag: d.Get("etag").(string),
+			})
+			if err != nil {
+				return err
+			}
+			return common.StructToData(data, s, d)
+		},
+		Update: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			var ruleSetUpdateReq iam.UpdateRuleSetRequest
+			common.DataToStructPointer(d, s, &ruleSetUpdateReq.RuleSet)
+			ruleSetUpdateReq.Name = ruleSetUpdateReq.RuleSet.Name
+			// etag should already be present
+			updatedEtag, err := handleConflictAndUpdate(ctx, c, ruleSetUpdateReq)
+			if err != nil {
+				return err
+			}
+			d.Set("etag", updatedEtag)
+			return nil
+		},
+		Delete: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			// we remove all grant rules. Account admins will still be able to update rule set
+			updatedEtag, err := handleConflictAndUpdate(ctx, c, iam.UpdateRuleSetRequest{
+				Name: d.Id(),
+				RuleSet: iam.RuleSetUpdateRequest{
+					Name: d.Id(),
+					Etag: d.Get("etag").(string),
+				},
+			})
+			if err != nil {
+				return err
+			}
+			d.Set("etag", updatedEtag)
```

Comment:
I'm pretty sure this is not necessary because resource state is removed after deleting a resource.

Reply:
Got it, will remove!